firefox-ios/Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -39,14 +38,17 @@ class TabScrollingController: NSObject,
             self.scrollView?.addGestureRecognizer(panGesture)
             scrollView?.delegate = self
             scrollView?.keyboardDismissMode = .onDrag
-            configureRefreshControl()
-            tab?.onLoading = { [weak self] in
-                guard let self else { return }
-                if tabIsLoading() {
-                    pullToRefreshView?.stopObservingContentScroll()
-                    pullToRefreshView?.removeFromSuperview()
-                } else {
-                    configureRefreshControl()
+            configureRefreshControl(isEnabled: true)
+            if isPullToRefreshRefactorEnabled {
+                tab?.onLoading = { [weak self] in
+                    guard let self else { return }
+                    // If we are in the home page delete pull to refresh so it doesn't show on the background
+                    if tabIsLoading() || (tab?.isFxHomeTab ?? false) {
+                        pullToRefreshView?.stopObservingContentScroll()
+                        pullToRefreshView?.removeFromSuperview()
+                    } else {
+                        configureRefreshControl(isEnabled: true)
+                    }
```

Comment:
Just to make this easier to read a bit, could we have this in it's own `func` ? Like `configurePullToRefresh` or some name that makes sense

Reply:
Yes sure 👍

Reply:
Yes sure, i separated this into another method, i didn't choose something like configPullToRefresh since we already had one. i put `setupTabOnLoadingCallback()` let me know what you think. Thanks @lmarceau 😃